File: sorting/mergesort_test.py
from sorting.mergesort import merge, merge_sort


def test_merge():
    left = [3, 4, 5]
    right = [5, 8, 12]
    expected = [3, 4, 5, 5, 8, 12]
    given = merge(left, right)
    assert given == expected


def test_mergesort():
    arr = [3, 2, 6, 1, 4, 6, 100, -1]
    given = merge_sort(arr)
    expected = [-1, 1, 2, 3, 4, 6, 6, 100]
    assert given == expected

    arr = []
    given = merge_sort(arr)
    expected = []
    assert given == expected

    arr = [2, 1]
    given = merge_sort(arr)
    expected = [1, 2]
    assert given == expected

File: sorting/mergesort.test.js
const mergeSort = require('./mergesort')

test('merge sort full', () => {
    const arr = [3, 2, 6, 1, 4, 6, 100, -1]
    const given = mergeSort(arr)
    const expected = [-1, 1, 2, 3, 4, 6, 6, 100]
    expect(given).toEqual(expected)
})

test('merge sort empty', () => {
    const arr = []
    const given = mergeSort(arr)
    const expected = []
    expect(given).toEqual(expected)
})

test('merge sort minimal', () => {
    const arr = [2, 1]
    const given = mergeSort(arr)
    const expected = [1, 2]
    expect(given).toEqual(expected)
})